From aaf1a9878e5d43da954ed91e91c9db3c37f80460 Mon Sep 17 00:00:00 2001
From: David 'Digit' Turner <digit@google.com>
Date: Thu, 2 May 2013 14:10:19 +0200
Subject: Support building with GCC 4.6

---
 include/__config | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/include/__config b/include/__config
index ca850d3..2223a08 100644
--- a/include/__config
+++ b/include/__config
@@ -370,6 +370,10 @@ namespace std {
 #define _LIBCPP_HAS_NO_STATIC_ASSERT
 #endif
 
+#if _GNUC_VER < 407
+#define _LIBCPP_HAS_NO_DEFAULTED_FUNCTIONS
+#endif
+
 #if _GNUC_VER < 404
 #define _LIBCPP_HAS_NO_ADVANCED_SFINAE
 #define _LIBCPP_HAS_NO_DECLTYPE
@@ -386,6 +390,10 @@ namespace std {
 
 #endif  // __GXX_EXPERIMENTAL_CXX0X__
 
+#if _GNUC_VER > 403
+#define _LIBCP_HAS_IS_BASE_OF
+#endif
+
 #define _LIBCPP_BEGIN_NAMESPACE_STD namespace std { namespace _LIBCPP_NAMESPACE {
 #define _LIBCPP_END_NAMESPACE_STD  } }
 #define _VSTD std::_LIBCPP_NAMESPACE
-- 
1.7.12.146.g16d26b1

