From 1c99272d860d0d753803c156d60d20a07ec27255 Mon Sep 17 00:00:00 2001
From: David 'Digit' Turner <digit@google.com>
Date: Thu, 2 May 2013 14:11:11 +0200
Subject: include/regex: Support building with GCC 4.6

---
 include/regex | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/include/regex b/include/regex
index 19e08b1..2417dc9 100644
--- a/include/regex
+++ b/include/regex
@@ -933,7 +933,7 @@ public:
     typedef locale                  locale_type;
     typedef ctype_base::mask        char_class_type;
 
-    static const char_class_type __regex_word = 0x80;
+    static const char_class_type __regex_word;
 private:
     locale __loc_;
     const ctype<char_type>* __ct_;
@@ -1008,9 +1008,13 @@ private:
     int __value(wchar_t __ch, int __radix) const;
 };
 
+template <class _CharT> const typename regex_traits<_CharT>::char_class_type regex_traits<_CharT>::__regex_word = 0x80;
+
+#if 0
 template <class _CharT>
 const typename regex_traits<_CharT>::char_class_type
 regex_traits<_CharT>::__regex_word;
+#endif
 
 template <class _CharT>
 regex_traits<_CharT>::regex_traits()
-- 
1.7.12.146.g16d26b1

