From 410b59fd1cbf0e94933ffe09850e79344b899390 Mon Sep 17 00:00:00 2001
From: David 'Digit' Turner <digit@google.com>
Date: Thu, 2 May 2013 14:11:53 +0200
Subject: [PATCH 3/7] src/new.cpp: Support building with GAbi++

---
 src/new.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/new.cpp b/src/new.cpp
index fa0331a..b08b798 100644
--- a/src/new.cpp
+++ b/src/new.cpp
@@ -150,13 +150,13 @@ operator delete[] (void* ptr, const std::nothrow_t&) _NOEXCEPT
 namespace std
 {
 
-#ifndef __GLIBCXX__
+#if !defined(__GLIBCXX__) && !defined(GABIXX_LIBCXX)
 const nothrow_t nothrow = {};
 #endif
 
 #ifndef _LIBCPPABI_VERSION
 
-#ifndef __GLIBCXX__
+#if !defined(__GLIBCXX__) && !defined(GABIXX_LIBCXX)
 
 new_handler
 set_new_handler(new_handler handler) _NOEXCEPT
-- 
1.8.4

