From cc35f95a952b52b0f9c5291c3bb044161e21d7b6 Mon Sep 17 00:00:00 2001
From: David 'Digit' Turner <digit@google.com>
Date: Thu, 2 May 2013 14:11:53 +0200
Subject: src/new.cpp: Support building with GAbi++

---
 src/new.cpp | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/src/new.cpp b/src/new.cpp
index b23a516..cf1d3c0 100644
--- a/src/new.cpp
+++ b/src/new.cpp
@@ -146,8 +146,9 @@ namespace std
 
 const nothrow_t nothrow = {};
 
-#ifndef _LIBCPPABI_VERSION
+#if !defined(_LIBCPPABI_VERSION)
 
+#if !defined(GABIXX_LIBCXX)
 new_handler
 set_new_handler(new_handler handler) _NOEXCEPT
 {
@@ -159,8 +160,9 @@ get_new_handler() _NOEXCEPT
 {
     return __sync_fetch_and_add(&__new_handler, (new_handler)0);
 }
+#endif
 
-#ifndef LIBCXXRT
+#if !defined(LIBCXXRT)
 
 bad_alloc::bad_alloc() _NOEXCEPT
 {
@@ -176,7 +178,7 @@ bad_alloc::what() const _NOEXCEPT
     return "std::bad_alloc";
 }
 
-#endif //LIBCXXRT
+#endif // !LIBCXXRT
 
 bad_array_new_length::bad_array_new_length() _NOEXCEPT
 {
@@ -192,7 +194,7 @@ bad_array_new_length::what() const _NOEXCEPT
     return "bad_array_new_length";
 }
 
-#endif
+#endif // !_LIBCPPABI_VERSION && !GABIXX_LIBCXX
 
 void
 __throw_bad_alloc()
-- 
1.7.12.146.g16d26b1

