From 71ac98ee6094e7f251df439b705f7adc6bce3869 Mon Sep 17 00:00:00 2001
From: Logan Chien <loganchien@google.com>
Date: Wed, 26 Sep 2012 14:23:44 +0800
Subject: [PATCH 19/20] Remove trailing whitespace in unwind.h

---
 lib/Headers/unwind.h |   40 ++++++++++++++++++++--------------------
 1 files changed, 20 insertions(+), 20 deletions(-)

diff --git a/lib/Headers/unwind.h b/lib/Headers/unwind.h
index a065920..bd8b73a 100644
--- a/lib/Headers/unwind.h
+++ b/lib/Headers/unwind.h
@@ -79,26 +79,26 @@ typedef enum {
 
 #ifdef __arm__
 
-typedef enum { 
-  _UVRSC_CORE = 0,        /* integer register */ 
-  _UVRSC_VFP = 1,         /* vfp */ 
-  _UVRSC_WMMXD = 3,       /* Intel WMMX data register */ 
-  _UVRSC_WMMXC = 4        /* Intel WMMX control register */ 
-} _Unwind_VRS_RegClass; 
-
-typedef enum { 
-  _UVRSD_UINT32 = 0,  
-  _UVRSD_VFPX = 1,  
-  _UVRSD_UINT64 = 3,  
-  _UVRSD_FLOAT = 4,  
-  _UVRSD_DOUBLE = 5 
-} _Unwind_VRS_DataRepresentation; 
-
-typedef enum { 
-  _UVRSR_OK = 0,  
-  _UVRSR_NOT_IMPLEMENTED = 1,  
-  _UVRSR_FAILED = 2  
-} _Unwind_VRS_Result; 
+typedef enum {
+  _UVRSC_CORE = 0,        /* integer register */
+  _UVRSC_VFP = 1,         /* vfp */
+  _UVRSC_WMMXD = 3,       /* Intel WMMX data register */
+  _UVRSC_WMMXC = 4        /* Intel WMMX control register */
+} _Unwind_VRS_RegClass;
+
+typedef enum {
+  _UVRSD_UINT32 = 0,
+  _UVRSD_VFPX = 1,
+  _UVRSD_UINT64 = 3,
+  _UVRSD_FLOAT = 4,
+  _UVRSD_DOUBLE = 5
+} _Unwind_VRS_DataRepresentation;
+
+typedef enum {
+  _UVRSR_OK = 0,
+  _UVRSR_NOT_IMPLEMENTED = 1,
+  _UVRSR_FAILED = 2
+} _Unwind_VRS_Result;
 
 _Unwind_VRS_Result _Unwind_VRS_Get(_Unwind_Context *context,
   _Unwind_VRS_RegClass regclass,
-- 
1.7.7.3

