From 48bcad97a1c8eaf9c7a0a79b452a35ffc27e3f52 Mon Sep 17 00:00:00 2001
From: Logan Chien <loganchien@google.com>
Date: Thu, 27 Sep 2012 12:34:47 +0800
Subject: [PATCH 21/23] Define _Unwind_GetIP for ARM.

---
 lib/Headers/unwind.h |    7 +++++++
 1 files changed, 7 insertions(+), 0 deletions(-)

diff --git a/lib/Headers/unwind.h b/lib/Headers/unwind.h
index 55ebd54..d6103c9 100644
--- a/lib/Headers/unwind.h
+++ b/lib/Headers/unwind.h
@@ -106,6 +106,13 @@ _Unwind_VRS_Result _Unwind_VRS_Get(struct _Unwind_Context *context,
   _Unwind_VRS_DataRepresentation representation,
   void *valuep);
 
+static inline uintptr_t _Unwind_GetIP(struct _Unwind_Context *context) {
+  uintptr_t ip;
+  _Unwind_VRS_Get(context, _UVRSC_CORE, 15, _UVRSD_UINT32, &ip);
+  ip &= ~(uintptr_t)0x1; /* remove thumb mode bit */
+  return ip;
+}
+
 #else
 
 uintptr_t _Unwind_GetIP(struct _Unwind_Context* context);
-- 
1.7.7.3

