From 8348a8c3e2d9fd5613b8dfca2201c941ed720d00 Mon Sep 17 00:00:00 2001
From: Logan Chien <loganchien@google.com>
Date: Thu, 27 Sep 2012 16:21:46 +0800
Subject: [PATCH 23/23] Look for libraries in <install-dir>/lib.

---
 lib/Driver/ToolChains.cpp |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/lib/Driver/ToolChains.cpp b/lib/Driver/ToolChains.cpp
index fc7a818..d1d7412 100644
--- a/lib/Driver/ToolChains.cpp
+++ b/lib/Driver/ToolChains.cpp
@@ -2103,6 +2103,8 @@ Linux::Linux(const Driver &D, const llvm::Triple &Triple, const ArgList &Args)
       addPathIfExists(LibPath + "/../" + GCCTriple.str() + "/lib", Paths);
       addPathIfExists(LibPath, Paths);
     }
+
+    addPathIfExists(LibPath + "/../" + GCCTriple.str() + "/lib", Paths);
   }
   addPathIfExists(SysRoot + "/lib", Paths);
   addPathIfExists(SysRoot + "/usr/lib", Paths);
-- 
1.7.7.3

