diff -Naur Driver.0/Tools.cpp Driver/Tools.cpp
--- Driver.0/Tools.cpp	2012-10-04 18:19:52.000000000 +0800
+++ Driver/Tools.cpp	2012-10-05 20:35:38.991983465 +0800
@@ -1179,6 +1179,7 @@
     Args.ClaimAllArgs(options::OPT_fno_cxx_exceptions);
     return;
   }
+  const bool isAndroid = Triple.getEnvironment() == llvm::Triple::ANDROID;
 
   // Exceptions are enabled by default.
   bool ExceptionsEnabled = true;
@@ -1200,7 +1201,7 @@
 
   // Exception tables and cleanups can be enabled with -fexceptions even if the
   // language itself doesn't support exceptions.
-  if (ExceptionsEnabled && DidHaveExplicitExceptionFlag)
+  if (ExceptionsEnabled && (DidHaveExplicitExceptionFlag || isAndroid))
     ShouldUseExceptionTables = true;
 
   // Obj-C exceptions are enabled by default, regardless of -fexceptions. This
