From 8af06f06691438b44e2bed6290fa08bc2a3209e1 Mon Sep 17 00:00:00 2001
From: David 'Digit' Turner <digit@google.com>
Date: Tue, 7 Feb 2012 14:20:10 +0100
Subject: Options brought in from core/combo/* for IA

---
 Makefile.in |   29 +++++++++++++++++++++++++++++
 1 files changed, 29 insertions(+), 0 deletions(-)

diff --git a/Makefile.in b/Makefile.in
index 0b73ff6..c7b5847 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -431,6 +431,35 @@ GCC_CONFIG_ARGS += $(GCC_CONFIG_LIBSTDCXX_V3) --disable-libssp \
 	--disable-tls --disable-libitm
 
 CFLAGS_FOR_TARGET += -DTARGET_POSIX_IO -fno-short-enums
+
+ifeq ($(target_cpu),x86)
+# Options brought in from core/combo/* for IA
+CFLAGS_FOR_TARGET += \
+	-DANDROID -D__ANDROID__ -Ulinux \
+	-fPIC -Wa,--noexecstack -m32 -fstack-protector \
+	-W -Wall -Werror=address -Werror=format-security -Werror=non-virtual-dtor -Werror=return-type \
+	-Werror=sequence-point -Winit-self -Wno-multichar -Wno-unused -Wpointer-arith -Wstrict-aliasing=2 \
+	-fexceptions -ffunction-sections -finline-functions \
+	-finline-limit=300 -fmessage-length=0 -fno-inline-functions-called-once \
+	-fno-strict-aliasing -frtti \
+	-fstrict-aliasing -funswitch-loops -funwind-tables \
+	-march=i686 -mtune=atom -mbionic -mfpmath=sse -mstackrealign -DUSE_SSE2
+
+LDFLAGS_FOR_TARGET += \
+	-m32 -O2 -g -fPIC \
+	-nostartfiles \
+	-Wl,-z,noexecstack -Wl,--gc-sections -nostdlib \
+	-fexceptions -frtti -fstrict-aliasing -ffunction-sections -finline-functions  \
+	-finline-limit=300 -fno-inline-functions-called-once \
+	-funswitch-loops -funwind-tables -mstackrealign \
+	-ffunction-sections -funwind-tables -fmessage-length=0 \
+	-march=atom -mstackrealign -mfpmath=sse -mbionic \
+	-Wno-multichar -Wl,-z,noexecstack -Werror=format-security -Wstrict-aliasing=2 \
+	-W -Wall -Wno-unused -Winit-self -Wpointer-arith -Werror=return-type -Werror=non-virtual-dtor \
+	-Werror=address -Werror=sequence-point \
+	-Werror=format-security -Wl,--no-undefined
+endif
+
 # ARM specific options.
 ifeq ($(target_cpu),arm)
 GCC_CONFIG_ARGS += --with-float=soft --with-fpu=vfp --with-arch=armv5te \
-- 
1.7.6.rc0

