diff -Naur llvm.0/llvm-3.1/tools/clang/lib/Driver/Tools.cpp llvm/llvm-3.1/tools/clang/lib/Driver/Tools.cpp
--- a/llvm-3.1/tools/clang/lib/Driver/Tools.cpp	2012-09-17 19:43:56.000000000 -0700
+++ b/llvm-3.1/tools/clang/lib/Driver/Tools.cpp	2012-09-25 15:29:22.000048999 -0700
@@ -1012,7 +1012,7 @@
       if (getToolChain().getArch() == llvm::Triple::x86_64)
         CPUName = "x86-64";
       else if (getToolChain().getArch() == llvm::Triple::x86)
-        CPUName = "pentium4";
+	CPUName = isAndroid? "i686" : "pentium4";
     }
   }
 
