diff -Naur llvm.0/llvm-3.1/tools/clang/lib/Basic/Targets.cpp llvm/llvm-3.1/tools/clang/lib/Basic/Targets.cpp
--- a/llvm-3.1/tools/clang/lib/Basic/Targets.cpp	2012-08-28 00:45:53.000000000 -0700
+++ b/llvm-3.1/tools/clang/lib/Basic/Targets.cpp	2012-09-25 16:28:53.080650937 -0700
@@ -1398,8 +1398,11 @@
     : TargetInfo(triple), SSELevel(NoSSE), MMX3DNowLevel(NoMMX3DNow),
       HasAES(false), HasLZCNT(false), HasBMI(false), HasBMI2(false),
       HasPOPCNT(false), HasFMA4(false), CPU(CK_Generic) {
+    llvm::Triple Triple = llvm::Triple(triple);
+    const bool isAndroid = Triple.getEnvironment() == llvm::Triple::ANDROID;	 
     BigEndian = false;
-    LongDoubleFormat = &llvm::APFloat::x87DoubleExtended;
+    if (!isAndroid)
+        LongDoubleFormat = &llvm::APFloat::x87DoubleExtended;
   }
   virtual unsigned getFloatEvalMethod() const {
     // X87 evaluates with 80 bits "long double" precision.
@@ -2207,13 +2210,21 @@
 class X86_32TargetInfo : public X86TargetInfo {
 public:
   X86_32TargetInfo(const std::string& triple) : X86TargetInfo(triple) {
+    llvm::Triple Triple = llvm::Triple(triple);
+    const bool isAndroid = Triple.getEnvironment() == llvm::Triple::ANDROID;	 
     DoubleAlign = LongLongAlign = 32;
-    LongDoubleWidth = 96;
+    if (!isAndroid)
+        LongDoubleWidth = 96;
     LongDoubleAlign = 32;
     SuitableAlign = 128;
-    DescriptionString = "e-p:32:32:32-i1:8:8-i8:8:8-i16:16:16-i32:32:32-"
-                        "i64:32:64-f32:32:32-f64:32:64-v64:64:64-v128:128:128-"
-                        "a0:0:64-f80:32:32-n8:16:32-S128";
+    if (!isAndroid)
+        DescriptionString = "e-p:32:32:32-i1:8:8-i8:8:8-i16:16:16-i32:32:32-"
+                            "i64:32:64-f32:32:32-f64:32:64-v64:64:64-v128:128:128-"
+                            "a0:0:64-f80:32:32-n8:16:32-S128";
+    else
+        DescriptionString = "e-p:32:32:32-i1:8:8-i8:8:8-i16:16:16-i32:32:32-"
+                            "i64:32:64-f32:32:32-f64:32:64-v64:64:64-v128:128:128-"
+                            "a0:0:64-f64:32:32-n8:16:32-S128";
     SizeType = UnsignedInt;
     PtrDiffType = SignedInt;
     IntPtrType = SignedInt;
