diff -Naur llvm-3.1.0/tools/clang/lib/Driver/ToolChains.cpp llvm-3.1/tools/clang/lib/Driver/ToolChains.cpp
--- a/llvm-3.1/tools/clang/lib/Driver/ToolChains.cpp	2012-09-27 02:55:07.000000000 -0700
+++ b/llvm-3.1/tools/clang/lib/Driver/ToolChains.cpp	2012-09-27 02:56:02.000000000 -0700
@@ -1118,7 +1118,6 @@
   llvm::Triple MultiarchTriple
     = TargetTriple.isArch32Bit() ? TargetTriple.get64BitArchVariant()
                                  : TargetTriple.get32BitArchVariant();
-  llvm::Triple::ArchType TargetArch = TargetTriple.getArch();
   // The library directories which may contain GCC installations.
   SmallVector<StringRef, 4> CandidateLibDirs, CandidateMultiarchLibDirs;
   // The compatible GCC triples for this particular architecture.
@@ -1156,7 +1155,7 @@
       if (!llvm::sys::fs::exists(LibDir))
         continue;
       for (unsigned k = 0, ke = CandidateTripleAliases.size(); k < ke; ++k)
-        ScanLibDirForGCCTriple(TargetArch, LibDir, CandidateTripleAliases[k]);
+        ScanLibDirForGCCTriple(TargetTriple, LibDir, CandidateTripleAliases[k]);
     }
     for (unsigned j = 0, je = CandidateMultiarchLibDirs.size(); j < je; ++j) {
       const std::string LibDir
@@ -1165,7 +1164,7 @@
         continue;
       for (unsigned k = 0, ke = CandidateMultiarchTripleAliases.size(); k < ke;
            ++k)
-        ScanLibDirForGCCTriple(TargetArch, LibDir,
+        ScanLibDirForGCCTriple(TargetTriple, LibDir,
                                CandidateMultiarchTripleAliases[k],
                                /*NeedsMultiarchSuffix=*/true);
     }
@@ -1310,8 +1309,10 @@
 }
 
 void Generic_GCC::GCCInstallationDetector::ScanLibDirForGCCTriple(
-    llvm::Triple::ArchType TargetArch, const std::string &LibDir,
+    const llvm::Triple &TargetTriple, const std::string &LibDir,
     StringRef CandidateTriple, bool NeedsMultiarchSuffix) {
+  const llvm::Triple::ArchType TargetArch = TargetTriple.getArch();
+  const bool IsAndroid = TargetTriple.getEnvironment() == llvm::Triple::ANDROID;
   // There are various different suffixes involving the triple we
   // check for. We also record what is necessary to walk from each back
   // up to the lib directory.
@@ -1352,9 +1353,17 @@
       // *if* there is a subdirectory of the right name with crtbegin.o in it,
       // we use that. If not, and if not a multiarch triple, we look for
       // crtbegin.o without the subdirectory.
-      StringRef MultiarchSuffix
-        = (TargetArch == llvm::Triple::x86_64 ||
-           TargetArch == llvm::Triple::ppc64) ? "/64" : "/32";
+      StringRef MultiarchSuffix;
+      if (!IsAndroid) {
+        MultiarchSuffix = (TargetArch == llvm::Triple::x86_64 ||
+                           TargetArch == llvm::Triple::ppc64) ? "/64" : "/32";
+      } else {
+        StringRef TargetTripleStr = TargetTriple.str();
+        if (TargetTripleStr.startswith("armv7") || TargetTripleStr.startswith("thumbv7"))
+          MultiarchSuffix = (TargetArch == llvm::Triple::thumb)? "/armv7-a/thumb" : "/armv7-a";
+        else
+          MultiarchSuffix = (TargetArch == llvm::Triple::thumb)? "/thumb" : "/32";
+      }
       if (llvm::sys::fs::exists(LI->path() + MultiarchSuffix + "/crtbegin.o")) {
         GCCMultiarchSuffix = MultiarchSuffix.str();
       } else {
diff -Naur llvm-3.1.0/tools/clang/lib/Driver/ToolChains.h llvm-3.1/tools/clang/lib/Driver/ToolChains.h
--- a/llvm-3.1/tools/clang/lib/Driver/ToolChains.h	2012-08-30 03:12:07.000000000 -0700
+++ b/llvm-3.1/tools/clang/lib/Driver/ToolChains.h	2012-09-27 02:56:09.000000000 -0700
@@ -110,7 +110,7 @@
       SmallVectorImpl<StringRef> &MultiarchLibDirs,
       SmallVectorImpl<StringRef> &MultiarchTripleAliases);
 
-    void ScanLibDirForGCCTriple(llvm::Triple::ArchType TargetArch,
+    void ScanLibDirForGCCTriple(const llvm::Triple &TargetTriple,
                                 const std::string &LibDir,
                                 StringRef CandidateTriple,
                                 bool NeedsMultiarchSuffix = false);
