Index: mingw-w64-headers/crt/stdio.h
===================================================================
--- mingw-w64-headers/crt/stdio.h	(revision 5158)
+++ mingw-w64-headers/crt/stdio.h	(working copy)
@@ -323,23 +323,14 @@
   return __mingw_vsprintf( __stream, __format, __local_argv );
 }
 
-__mingw_ovr
+extern
 __attribute__ ((__format__ (gnu_printf, 2, 3))) __attribute__((nonnull (1,2)))
-int asprintf(char **__ret, const char *__format, ...)
-{
-  register int __retval;
-  __builtin_va_list __local_argv; __builtin_va_start( __local_argv, __format );
-  __retval = __mingw_vasprintf( __ret, __format, __local_argv );
-  __builtin_va_end( __local_argv );
-  return __retval;
-}
+int asprintf(char **__ret, const char *__format, ...);
 
-__mingw_ovr
+extern
 __attribute__ ((__format__ (gnu_printf, 2, 0))) __attribute__((nonnull (1,2)))
-int vasprintf(char **__ret, const char *__format, __builtin_va_list __local_argv)
-{
-  return __mingw_vasprintf( __ret, __format, __local_argv );
-}
+int vasprintf(char **__ret, const char *__format, __builtin_va_list __local_argv);
+
 /* #ifndef __NO_ISOCEXT */  /* externs in libmingwex.a */
 __mingw_ovr
 __attribute__((__format__ (gnu_printf, 3, 4))) __MINGW_ATTRIB_NONNULL(3)
Index: mingw-w64-crt/stdio/vasprintf.c
===================================================================
--- mingw-w64-crt/stdio/vasprintf.c	(revision 5158)
+++ mingw-w64-crt/stdio/vasprintf.c	(working copy)
@@ -22,3 +22,7 @@
   return len;
 }
 
+int vasprintf(char **__ret, const char *__format, __builtin_va_list __local_argv)
+{
+  return __mingw_vasprintf( __ret, __format, __local_argv );
+}
Index: mingw-w64-crt/stdio/asprintf.c
===================================================================
--- mingw-w64-crt/stdio/asprintf.c	(revision 5158)
+++ mingw-w64-crt/stdio/asprintf.c	(working copy)
@@ -29,3 +29,11 @@
   return len;
 }
 
+int asprintf(char **__ret, const char *__format, ...)
+{
+  register int __retval;
+  __builtin_va_list __local_argv; __builtin_va_start( __local_argv, __format );
+  __retval = __mingw_vasprintf( __ret, __format, __local_argv );
+  __builtin_va_end( __local_argv );
+  return __retval;
+}
