diff -burN gcc-a/gcc-4.4.3/libstdc++-v3/src/Makefile.am gcc-b/gcc-4.4.3/libstdc++-v3/src/Makefile.am
--- gcc-a/gcc-4.4.3/libstdc++-v3/src/Makefile.am	2011-10-10 17:32:21.983349054 +0200
+++ gcc-b/gcc-4.4.3/libstdc++-v3/src/Makefile.am	2011-10-10 17:10:06.474674579 +0200
@@ -206,7 +206,7 @@
 	$(top_builddir)/libsupc++/libsupc++convenience.la
 
 libstdc___la_LDFLAGS = \
-	-version-info $(libtool_VERSION) ${version_arg} -lm 
+	-avoid-version ${version_arg} -lm 
 
 # Use special rules for the deprecated source files so that they find
 # deprecated include files.
diff -burN gcc-a/gcc-4.4.3/libstdc++-v3/src/Makefile.in gcc-b/gcc-4.4.3/libstdc++-v3/src/Makefile.in
--- gcc-a/gcc-4.4.3/libstdc++-v3/src/Makefile.in	2011-10-10 17:32:21.983349054 +0200
+++ gcc-b/gcc-4.4.3/libstdc++-v3/src/Makefile.in	2011-10-10 17:10:06.514673034 +0200
@@ -350,7 +350,7 @@
 AM_CPPFLAGS = $(GLIBCXX_INCLUDES)
 
 # Cross compiler support.
-toolexeclib_LTLIBRARIES = libstdc++.la
+toolexeclib_LTLIBRARIES = libstdc++.la libgnustl_shared.la
 @ENABLE_SYMVERS_DARWIN_TRUE@@ENABLE_SYMVERS_TRUE@version_arg = -Wl,-exported_symbols_list,libstdc++-symbols.explist
 @ENABLE_SYMVERS_FALSE@version_arg = 
 @ENABLE_SYMVERS_GNU_NAMESPACE_TRUE@@ENABLE_SYMVERS_TRUE@version_arg = -Wl,--version-script=libstdc++-symbols.ver
@@ -454,7 +454,7 @@
 	$(top_builddir)/libsupc++/libsupc++convenience.la
 
 libstdc___la_LDFLAGS = \
-	-version-info $(libtool_VERSION) ${version_arg} -lm 
+	-avoid-version ${version_arg} -lm 
 
 
 # Use special rules for the deprecated source files so that they find
@@ -576,6 +576,9 @@
 libstdc++.la: $(libstdc___la_OBJECTS) $(libstdc___la_DEPENDENCIES) 
 	$(CXXLINK) -rpath $(toolexeclibdir) $(libstdc___la_LDFLAGS) $(libstdc___la_OBJECTS) $(libstdc___la_LIBADD) $(LIBS)
 
+libgnustl_shared.la: $(libstdc___la_OBJECTS) $(libstdc___la_DEPENDENCIES) 
+	$(CXXLINK) -rpath $(toolexeclibdir) $(libstdc___la_LDFLAGS) $(libstdc___la_OBJECTS) $(libstdc___la_LIBADD) $(LIBS)
+
 mostlyclean-compile:
 	-rm -f *.$(OBJEXT)
 
